src/main/java/org/gridsuite/modification/server/modifications/LineModification.java

```diff
@@ -99,61 +97,91 @@ private void modifyCharacteristics(Line line, LineModificationInfos lineModifica
             .withDefaultMessage("Characteristics")
             .withSeverity(TypedValue.INFO_SEVERITY)
             .build());
-        if (lineModificationInfos.getSeriesResistance() != null) {
-            characteristicsReporter.report(ModificationUtils.getInstance().buildModificationReportWithIndentation(line.getR(),
-                    lineModificationInfos.getSeriesResistance().getValue(), "Series resistance", 1));
-            line.setR(lineModificationInfos.getSeriesResistance().getValue());
+        Double seriesResistance = lineModificationInfos.getSeriesResistance() != null
+                ? lineModificationInfos.getSeriesResistance().getValue()
+                : null;
+        if (seriesResistance != null) {
+            characteristicsReporter
+                    .report(ModificationUtils.getInstance().buildModificationReportWithIndentation(line.getR(),
+                            seriesResistance, "Series resistance", 1));
+            line.setR(seriesResistance);
         }
-        if (lineModificationInfos.getSeriesReactance() != null) {
-            characteristicsReporter.report(ModificationUtils.getInstance().buildModificationReportWithIndentation(line.getX(),
-                    lineModificationInfos.getSeriesReactance().getValue(), "Series reactance", 1));
-            line.setX(lineModificationInfos.getSeriesReactance().getValue());
+
+        Double seriesReactance = lineModificationInfos.getSeriesReactance() != null
+                ? lineModificationInfos.getSeriesReactance().getValue()
+                : null;
+        if (seriesReactance != null) {
+            characteristicsReporter
+                    .report(ModificationUtils.getInstance().buildModificationReportWithIndentation(line.getX(),
+                            seriesReactance, "Series reactance", 1));
+            line.setX(seriesReactance);
         }
-        if (lineModificationInfos.getShuntConductance1() != null
-                || lineModificationInfos.getShuntSusceptance1() != null) {
+
+        modifySide1Characteristics(line, lineModificationInfos, characteristicsReporter);
+        modifySide2Characteristics(line, lineModificationInfos, characteristicsReporter);
+
+    }
+
+    private void modifySide1Characteristics(Line line, LineModificationInfos lineModificationInfos,
+            Reporter characteristicsReporter) {
+        Double shuntConductance1 = lineModificationInfos.getShuntConductance1() != null
+                ? lineModificationInfos.getShuntConductance1().getValue()
+                : null;
+        Double shuntSusceptance1 = lineModificationInfos.getShuntSusceptance1() != null
+                ? lineModificationInfos.getShuntSusceptance1().getValue()
+                : null;
+        if (shuntConductance1 != null
+                || shuntSusceptance1 != null) {
```

Comment:
i think we can check them one by one:
if (lineModificationInfos.getShuntConductance1() != null)
{......}
if (lineModificationInfos.getShuntSusceptance1() != null)
{......}